### PR TITLE
Fix span of nodes with decorators, parameter properties, and class members with accessibility

### PR DESCRIPTION
**Pull request by dsherret**
* Spans of class members and classes should include decorators.
* Fix hi position for class parameter properties.
* Fix span of members with accessibility.

### CODE REVIEW

**Comment by dsherret** [comment]
I think the spans of function nodes with decorators might be still incorrect (not going to bother with them for this PR as they're not supported in typescript).

File: ecmascript/parser/src/parser/class_and_fn.rs [hunk 1]
```diff
@@ -364,19 +363,18 @@ impl<'a, I: Tokens> Parser<'a, I> {
...
     #[allow(clippy::cognitive_complexity)]
     fn parse_class_member_with_is_static(
         &mut self,
+        start: BytePos,
         accessibility: Option<Accessibility>,
         static_token: Option<Span>,
         decorators: Vec<Decorator>,
     ) -> PResult<'a, ClassMember> {
         let is_static = static_token.is_some();
-        let start = static_token.map(|s| s.lo()).unwrap_or(cur_pos!());
```
Comment (dsherret) [inline ecmascript/parser/src/parser/class_and_fn.rs]: This start did not include decorators or the accessibility modifier start.

Comment (kdy1) [reply ecmascript/parser/src/parser/class_and_fn.rs]: You are right. I think I forgot to update it when I added a typescript parser and decorators.

Comment (dsherret) [reply ecmascript/parser/src/parser/class_and_fn.rs]: Makes sense. Yeah, it's no problem—very easy to fix 🙂

**Pull request merged**